### ISSUE TITLE
[Experimental] Support Kubernetes MCS

### ISSUE THREAD

**Issue by nmittler** [istio/enhancements#9]
[//]: # (The syntax preceeding this line is a comment marker used to help guide the issue author with submission of an issue)
[//]: # (to github. Unlike HTML comments commonly used throughout istio.io documentation, this comment will not be rendered)
[//]: # (by github. Comments must be separated by carriage return preceding and concluding the text and be a single line.)

[//]: # (This page lists the requirements for an experimental feature. Completion of this template enables Istio work groups)
[//]: # (to collect information on potential new functionality. This template should be completed before users are exposed to)
[//]: # (any new experimental feature. Please complete this template during development.)

[//]: # (The feature implementation section must be completed before submission of the issue.)

# Feature:

[//]: # (All information in this section is mandatory for new issue submission.)

**Feature name:**

[//]: # (The name of the feature, e.g. Multiple control planes)

Support Kubernetes MCS

**Primary lead(s):**

[//]: # (The primary lead or leads responsible for the feature. These individuals serve as a point of contact for the feature.)

Nathan Mittler (@nmittler)

**Short description:**

[//]: # (A short description of the feature. One or two sentences maximum.)

Add support for [Kubernetes Multi-cluster Services (MCS)](https://github.com/kubernetes/enhancements/tree/master/keps/sig-multicluster/1645-multi-cluster-services-api) to Istio.

# Requirements:

[//]: # (All information in this section is mandatory for promotion. Please edit the issue to modify the links in this)
[//]: # (section.)

- [ ] Export all services by default, but allow the default to be configurable.
- [ ] Allow services to opt-in/out of auto-export.
- [ ] Process ServiceImport and associated EndpointSlices.
- [ ] Optionally generate ServiceImport and EndpointSlices where appropriate for each exported service.
- [ ] Support multiple networks. Typically this means that endpoints in remote networks are reached via a gateway.
- [ ] Support even load balancing across clusters. For example, if both Cluster X and Y have 2 endpoints for Service A, traffic to Service A should be evenly load balanced across all 4 endpoints, regardless of whether the services are accessed directly or via gateways (as in the case of multi-network).

- [ ] [User stories](insert_your_link_here) reviewed in a work group meeting.

[//]: # (User stories are a way to communicate user value. User stories follow the style)
[//]: # (as a [type of user], I want [an action] so that [a benefit/a value]. Istio currently has no user)
[//]: # (story template. Maybe you can make one?)

[//]: # (User stories must be presented in a work group meeting. They need no approval and are later integrated)
[//]: # (into the RFCs, which do need approval for alpha. You may find value to negotiate within the work group where the)
[//]: # (user stories are presented to help clarify the user stories.)

- [x] [RFC Authored](https://docs.google.com/document/d/1K8hvQ83UcJ9a7U8oqXIefwr6pFJn-VBEi40Ak-fwQtk/edit#).

[//]: # (An RFC is mandatory to graduate to experimental. The RFC does not have to be reviewed in a work group)
[//]: # (meeting to graduate to experimental.)

- [ ] [Documentation](insert_your_link_here) for enabling and using the feature.

[//]: # (The documentation instructions may exist on the developer wiki or the team drive. They may include instructions)
[//]: # (for building running a `istioctl experimental command`, or using the preview profile,)
[//]: # (or any other relevant information.)

- [ ] [Feedback plan](insert_your_link_here).

[//]: # (This may include user feedback meetings, discuss.istio.io conversations, GitHub issues, or mailing lists.)

- [x] Disabled by default.

- [x] No impact on performance when the feature is disabled.

**Comment by brian-avery** [comment]
Migrated to https://github.com/brian-avery/enhancements/blob/migrateFeaturesToFiles/features/Kubernetes_MCS.md as part of #26
